src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_command_factory.cc

```diff
@@ -74,125 +144,125 @@ CommandSharedPtr RCCommandFactory::CreateCommand(
-    return CreateMobileCommand(message, source);
+    hmi_apis::messageType::eType message_type =
+        static_cast<hmi_apis::messageType::eType>(
+            (*message)[strings::params][strings::message_type].asInt());
+
+    hmi_apis::FunctionID::eType function_id =
+        static_cast<hmi_apis::FunctionID::eType>(
+            (*message)[strings::params][strings::function_id].asInt());
+
+    return get_hmi_creator_factory(function_id, message_type).create(message);
   }
 }
 
-CommandSharedPtr RCCommandFactory::CreateMobileCommand(
-    const app_mngr::commands::MessageSharedPtr& message,
-    app_mngr::commands::Command::CommandSource source) {
-  // TODO : rework factory same as in SDL commanf dactory
-  return CommandSharedPtr();
-
-  //  CommandSharedPtr command;
-  //  const int function_id =
-  //      (*message)[strings::params][strings::function_id].asInt();
-  //  LOG4CXX_DEBUG(logger_, "CreateMobileCommand function_id: " <<
-  //  function_id);
-  //  switch (function_id) {
-  //    case mobile_apis::FunctionID::ButtonPressID: {
-  //      if ((*message)[strings::params][strings::message_type] ==
-  //          static_cast<int>(application_manager::MessageType::kRequest)) {
-  //        command.reset(new commands::ButtonPressRequest(
-  //            message, app_manager_, allocation_manager_));
-  //      } else {
-  //        command.reset(new commands::ButtonPressResponse(message,
-  //        app_manager_));
-  //      }
-  //      break;
-  //    }
-  //    case mobile_apis::FunctionID::GetInteriorVehicleDataID: {
-  //      if ((*message)[strings::params][strings::message_type] ==
-  //          static_cast<int>(application_manager::MessageType::kRequest)) {
-  //        command.reset(
-  //            new commands::GetInteriorVehicleDataRequest(message,
-  //            app_manager_, allocation_manager_));
-  //      } else {
-  //        command.reset(new commands::GetInteriorVehicleDataResponse(
-  //            message, app_manager_));
-  //      }
-  //      break;
-  //    }
-  //    case mobile_apis::FunctionID::SetInteriorVehicleDataID: {
-  //      if ((*message)[strings::params][strings::message_type] ==
-  //          static_cast<int>(application_manager::MessageType::kRequest)) {
-  //        command.reset(
-  //            new commands::SetInteriorVehicleDataRequest(message,
-  //            app_manager_));
-  //      } else {
-  //        command.reset(new commands::SetInteriorVehicleDataResponse(
-  //            message, app_manager_));
-  //      }
-  //      break;
-  //    }
-  //    case mobile_apis::FunctionID::OnInteriorVehicleDataID: {
-  //      command.reset(new commands::OnInteriorVehicleDataNotification(
-  //          message, app_manager_));
-  //      break;
-  //    }
-  //    default: { break; }
-  //  }
-  //  return command;
+bool RCCommandFactory::IsAbleToProcess(
+    const int32_t function_id,
+    const application_manager::commands::Command::CommandSource message_source)
+    const {
```

Comment:
@BSolonenko doe not look like standart style. Pleease run checkstyle

Reply:
@LuxoftAKutsan checked in https://github.com/LuxoftAKutsan/sdl_core/pull/275/commits/5a296e130d1911c156d105e21d0ef00d4c9b97b1